jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpConnection.java

```diff
@@ -906,12 +914,16 @@ private void releaseChunk()
         @Override
         protected void onCompleteSuccess()
         {
+            if (_shutdownOut && !(_httpChannel.getRequest().getAttribute(HttpStream.UPGRADE_CONNECTION_ATTRIBUTE) instanceof Connection))
+                getEndPoint().shutdownOutput();
```

Comment:
Why is this needed here?
It is already taken care of in `HttpStreamOverHTTP1.succeeded()`.

Reply:
Please read issue and comments!

Reply:
That reason should be stated in a comment.